pages/api/course/[id]/index.ts

```diff
@@ -0,0 +1,31 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { HttpMethod } from "@lib/http-method";
+import { getCourseWithBooks } from "@lib/services/course";
+import { StatusCodes } from "http-status-codes";
+import { validate } from 'uuid';
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+  switch (req.method as HttpMethod) {
+    case HttpMethod.GET:
+      return getCourseWithBooksHandler(req, res);
+    default:
+      return res.status(StatusCodes.METHOD_NOT_ALLOWED).end();
+  }
+}
+
+async function getCourseWithBooksHandler(req: NextApiRequest, res: NextApiResponse) {
+  const { id } = req.query;
+  const courseID = id as string;
```

Comment:
Maybe follow Caleb's recommendation for a PR I recently had?

https://github.com/calebmech/Book-Bazar/pull/43#discussion_r779765456

Reply:
Good catch 👍